sdk/src/Core/Amazon.Runtime/Telemetry/Metrics/AsyncMeasurementHandle.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+namespace Amazon.Runtime.Telemetry.Metrics
+{
+    /// <summary>
+    /// Represents a handle for async measurements.
+    /// The handle is used to manage and stop the async recording of measurements.
+    /// </summary>
+    /// <typeparam name="T">The type of value being measured.</typeparam>
+    public abstract class AsyncMeasurementHandle<T> where T : struct
+    {
+        /// <summary>
+        /// Stops the async measurement.
+        /// </summary>
+        public abstract void Stop();
```

Comment:
Instead of having to call `Stop` methods should we use the pattern of making the type implement `IDisposable` and then we can rely on the dispose pattern?

Reply:
Actually I should remove this stop method, this is one of the discrepancies between OpenTelemtery specs and the .net implementation.
From OTel specs: https://opentelemetry.io/docs/specs/otel/metrics/api/#asynchronous-instrument-api:~:text=Where%20the%20API%20supports%20registration%20of%20callback%20functions%20after%20asynchronous%20instrumentation%20creation%2C%20the%20user%20MUST%20be%20able%20to%20undo%20registration%20of%20the%20specific%20callback%20after%20its%20registration%20by%20some%20means.

And I couldn't find a way to do the unregistration, we should map `AsyncMeasurementHandle` to `ObservableInstrument` and there is no way to stop it and we have to dispose the whole meter we used (which now implements IDisposable) based on this https://github.com/open-telemetry/opentelemetry-dotnet/issues/3460#issuecomment-1190364356

Will add a reference for the `Meter` instead of the stop method.